lib/elixir/lib/exception.ex

```diff
@@ -456,29 +458,36 @@ defmodule Exception do
       iex> Exception.format_file_line("foo", nil)
       "foo:"
 
-      iex> Exception.format_file_line(nil, nil)
+      iex> Exception.format_file_line(nil, "bar")
       ""
 
   """
-  def format_file_line(file, line) do
-    format_file_line(file, line, "")
-  end
-
-  defp format_file_line(file, line, suffix) do
-    if file do
-      if line && line != 0 do
-        "#{file}:#{line}:#{suffix}"
-      else
+  @spec format_file_line(term, term, term) :: String.t
+  def format_file_line(file, line, suffix \\ "") do
+    cond do
+      file in [nil, false, ""] ->
+        ""
+      line in [0, nil, false, ""] ->
         "#{file}:#{suffix}"
-      end
-    else
-      ""
+      true ->
+        "#{file}:#{line}:#{suffix}"
     end
   end
 
   defp format_location(opts) when is_list(opts) do
     format_file_line Keyword.get(opts, :file), Keyword.get(opts, :line), " "
   end
+
+  @doc false
+  # Make sure file name is not invalid, so we don't get an unexpected
+  # error from Path.relative_to_cwd/1
+  def sanitize_file_line(file, line, suffix \\ "") do
+    if is_bitstring(file) and file != "" do
```

Comment:
It'd be better to use `is_binary/1` check here, what is sill a bit restrictive since `Path.relative_to_cwd/1` accepts `:unicode.chardata` type.